docs/reporting.md

```diff
@@ -0,0 +1,163 @@
+# SoberSwag Reporting
+
+`SoberSwag::Reporting` is a new module that provides a more *composable* interface of sober swag types.
+Unlike the base version, it is *not* based on `dry-types`, instead using a simpler scheme.
```

Comment:
You mention this is mostly a drop-in replacement, but not entirely so. Could you add a section on upgrading with things to watch out for, such as the removal of `default` for attributes?

Reply:
b29219b